### DISCUSSION TITLE
Automatically generate testset generator from struct definition

### DISCUSSION THREAD

**Discussion by minecraft2048** [Seelengrab/Supposition.jl#38] [Design & Ideas]
I have this giant struct:
```
struct GiantStruct
    a::UInt16
    b::Int32
    ...
   [100 struct members with primitive type]
end
```

Currently I need to manually add the primitive generators one by one:

```
generate_GiantStruct = @composed GiantStruct(
      Data.Integers{UInt16}(),
      Data.Integers{UInt32}(),
      ...
      [repeat 100 times with all the other struct members]
)
```

Currently I copy pasted the generators, but that's really tedious. It will be useful if I can somehow feed in the struct definition and it will magically populate all the struct members with the primitive data generators.

Ideally it can also handle nested struct, but I have no idea on how to implement that

**Comment by Seelengrab** [comment]
Thanks for opening this! I think this should basically be covered by https://github.com/Seelengrab/Supposition.jl/discussions/21. I have a patch locally that implements a first stab at this (even recursively), but it can't yet handle `UnionAll` properly.

**Reply by minecraft2048** [reply]
Your patch is good enough for me, my structs don't have those weird `UnionAll` types. Or even strings or any variable length data structures
